histomicsui/web_client/panels/DrawWidget.js

```diff
@@ -170,6 +192,73 @@ var DrawWidget = Panel.extend({
         });
     },
 
+    /**
+     * Respond to a click on the "view" button by changing the
+     * viewer location and zoom level to focus on one annotation
+     */
+    viewElement(evt) {
+        const annot = this.collection._byId[$(evt.target).parent().attr('data-id')];
+        let points;
+        let pointAnnot = false;
+        switch (annot.get('type')) {
+            case 'point':
+                points = [annot.get('center')];
+                pointAnnot = true;
+                break;
+            case 'polyline':
+                points = annot.get('points');
+                break;
+            case 'rectangle':
+                points = convertRectangle(annot.attributes).coordinates[0];
+                break;
+            case 'ellipse':
+                points = convertEllipse(annot.attributes).coordinates[0];
+                break;
+            case 'circle':
+                points = convertCircle(annot.attributes).coordinates[0];
+                break;
+        }
+        const xCoords = points.map((point) => {
+            return point[0];
+        });
+        const yCoords = points.map((point) => {
+            return point[1];
+        });
+        const bounds = {
+            left: Math.min(...xCoords),
+            top: Math.min(...yCoords),
+            right: Math.max(...xCoords),
+            bottom: Math.max(...yCoords)
+        };
+        const map = this.parentView.viewer;
+        const originalZoomRange = map.zoomRange();
+        map.zoomRange({
+            min: Number.NEGATIVE_INFINITY,
+            max: Number.POSITIVE_INFINITY
+        });
+        const newView = pointAnnot ?
+                        {
+                            center: {
+                                x: bounds.left,
+                                y: bounds.top
+                        },
+                        zoom: false
+                        } :
+                        map.zoomAndCenterFromBounds(bounds, map.rotation());
+        map.zoomRange(originalZoomRange);
+        if (Math.abs(this.zoomWidget.zoomToMagnification(newView.zoom) * 0.2968 + 1.8666 - this.zoomWidget.zoomToMagnification(map.zoom())) <= 40 && map.zoom() < newView.zoom) {
```

Comment:
Where do the values of 0.2968 and 1.8666 come from?  Why is there a magnification of 40 limit (different files have different pixel-level zooms, so it should probably be image based)?

Reply:
The values of 0.2968 and 1.8666 are from the linear equation I use to move from the zoom level given by `zoomAndCenterFromBounds()`, which provides the zoom needed to fit the annotation in the frame exactly, to a zoom level that has the annotation fill up about a quarter of the screen.  I switched from zoom levels to magnification, so the transformation would be universal across images of different sizes.  By plotting the magnification levels that led annotations with the same shape but different sizes to occupy the same space on my screen, I realized that there is a linear relationship between magnification level and area on screen, resulting in the linear transformation.  As for the numbers themselves, they are based on the annotation size that I thought looked good, but they can be changed if you feel that the annotations are too zoomed in or not zoomed in enough.  I went with this method because I thought it would work well for annotations of different shapes, but I could also explore calculating the new camera bounds based on the annotation's bounds themselves.

As for the magnification limit, this isn't actually a limit on the magnification itself, it's a limit on how big the difference between the current magnification and the desired new magnification can be for the transition to happen without changing the zoom level at all.  In your comment on the issue you mentioned that, 

> we probably shouldn't change the zoom if the annotation would fit on the display and is more than some minimum size.

so this if statement is addressing that.  If the new annotation fits in the screen (`map.zoom() < newView.zoom`) and it wouldn't be too small (ie. the magnification levels aren't too different), then the zoom level won't change.  Similar to above, the 40 number is one that I decided on based on my own investigation and can be changed.  I could also try to find a way to check if the fraction of the screen covered by the annotation is too small (though that would also require a somewhat arbitrary cutoff point).

Reply:
The zoom level is in log2 pixel space, whereas the magnification is in linear pixel space, so this could just be changed to avoid the calls to `zoomToMagnfication` and do everything in the log2 space.  That is, maybe using `newView.zoom - 1.5` as the desired new zoom level and something like `if (newView.zoom - 1.5 - map.zoom() < 2) { newView.zoom = false; }`.  I haven't checked if that last `< 2` is a sensible value.

Reply:
That makes sense!  Based on what I've seen 0.5 is a better value and I kept the `Math.min` part in because if an annotation is slightly larger than the current one I thought the zoom should stay the same as well.

See 3fb6fd6